gonvme_tcp_fc.go

```diff
@@ -178,21 +216,169 @@ func (nvme *NVMeTCP) discoverNVMeTCPTargets(address string, login bool) ([]NVMeT
 
 	// TODO: Add optional login
 	// log into the target if asked
-	/*if login {
+	if login {
 		for _, t := range targets {
-			gonvme.PerformLogin(t)
+			err = nvme.NVMeTCPConnect(t)
+			if err != nil {
+				log.Infof("Error during NVMeTCP connect")
+			}
 		}
-	}*/
+	}
+
+	return targets, nil
+}
+
+// DiscoverNVMeFCTargets - runs nvme discovery and returns a list of NVMeFC targets.
+func (nvme *NVMe) DiscoverNVMeFCTargets(targetAddress string, login bool) ([]NVMeTarget, error) {
+	return nvme.discoverNVMeFCTargets(targetAddress, login)
+}
+
+func (nvme *NVMe) discoverNVMeFCTargets(targetAddress string, login bool) ([]NVMeTarget, error) {
+	// TODO: add injection check on address
+	// nvme discovery is done via nvme cli
+	// nvme discover -t fc -a traddr -w host_traddr
+	// where traddr = nn-<Target_WWNN>:pn-<Target_WWPN> and host_traddr = nn-<Initiator_WWNN>:pn-<Initiator_WWPN>
+
+	var out []byte
+	FCHostsInfo, err := nvme.getFCHostInfo()
+	if err != nil {
+		log.Infof("Error gathering NVMe/FC Hosts on the host side: %v", err)
+		return []NVMeTarget{}, nil
```

Comment:
err should be returned here?

Reply:
If the fc hosts are not present on the host side, then shouldn't it return an error